.github/workflows/ci-semantic-pull-request.yml

```diff
@@ -0,0 +1,87 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+name: Semantic Pull Request
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - edited
+      - synchronize
```

Comment:
@tisonkun the github docs says
```
synchronize: Triggered when a pull request's head branch is updated. For example, when the head branch is updated from the base branch, when new commits are pushed to the head branch, or when the base branch is changed.
```
This means that for every push on the contributor branch, on every commit on the master branch this action is triggered again unnecessarly.

As you know we're facing GH runners supply issues so I suggest to remove this trigger type

Reply:
Make sense. I notice some weird check status not report issue previously when I removed it at https://github.com/apache/pulsar/pull/17961#discussion_r990756680.

But given the functionality is ready now we can try to apply the optimization.

Reply:
So here it is: https://github.com/apache/pulsar/pull/17997